plugins/ida/ida_main.ml

```diff
@@ -129,35 +129,26 @@ let loader path =
             else code, Memmap.add data mem sec) in
   Project.Input.create arch path ~code ~data
 
-let main () =
+let main ida_path is_headless =
+  let register_source = register_source ida_path is_headless in
+  let loader = loader ida_path is_headless in
   register_source (module Rooter);
   register_source (module Symbolizer);
   register_source (module Reconstructor);
   Project.Input.register_loader name loader
 
-module Main = struct
-  open Cmdliner
-  let man = [
-    `S "DESCRIPTION";
-    `P "This plugin provides rooter, symbolizer and reconstuctor services.";
-    `P "If IDA instance is found on the machine, or specified by a
-  user, it will be queried for the specified information."
-
-  ]
-
-
-  let path : string option Term.t =
-    let doc = "Use IDA to extract symbols from file. \
-               You can optionally provide path to IDA executable,\
-               or executable name." in
-    Arg.(value & opt (some string) None & info ["path"] ~doc)
-
-  let info = Term.info name ~version ~doc ~man
-
-  let () =
-    let run = Term.(const main $ const ()) in
-    match Term.eval ~argv ~catch:false (run,info) with
-    | `Ok () -> ()
-    | `Help | `Version -> exit 0
-    | `Error _ -> exit 1
-end
+let () =
+  let () = Config.manpage [
+      `S "DESCRIPTION";
+      `P "This plugin provides rooter, symbolizer and reconstuctor services.";
+      `P "If IDA instance is found on the machine, or specified by a
+        user, it will be queried for the specified information."
+    ] in
+  let path =
+    let doc = "Path to IDA directory." in
+    Config.(param (some string) "path" ~default:None ~doc) in
```

Comment:
`None` causes it to default to `Ida_config.path` anyways. Basically, if we do not keep compile time constants baked into the library, then it causes a large amount of complication for plugin authors who might want to use just the library, and not the `ida` plugin. I have implemented this on a separate branch if you want to take a look: https://github.com/BinaryAnalysisPlatform/bap/compare/master...jaybosamiya:ida-path